pandas/core/internals/managers.py

```diff
@@ -602,16 +602,27 @@ def comp(s, regex=False):
                 for b in rb:
                     m = masks[i][b.mgr_locs.indexer]
                     convert = i == src_len
-                    result = b._replace_coerce(
-                        mask=m,
-                        to_replace=s,
-                        value=d,
-                        inplace=inplace,
-                        convert=convert,
-                        regex=regex,
-                    )
                     if m.any() or convert:
-                        new_rb = _extend_blocks(result, new_rb)
+                        if m.ndim > 1:
+                            to_replace = m.any(axis=1)
+                            subblocks = [
```

Comment:
here you should insead use _split_op_result which creates the individual split blocks

this is too nested and not understandable.